tests/phpunit/tests/html/wpHtmlTagProcessor.php

```diff
@@ -0,0 +1,1960 @@
+				'<script class="d-md-none"></title></script><div></div>',
+			),
+
+			'Complex script with many parsing states'    => array(
+				'<script class="d-md-none"><!--<script>--><scRipt><span><!--<span><Script</script>--></scripT><div></div>-->',
+			),
+		);
+	}
+
+	/**
+	 * @ticket 56299
+	 *
+	 * @covers WP_HTML_Tag_Processor::next_tag
+	 *
+	 * @dataProvider data_next_tag_ignores_contents_of_rcdata_tag
+	 *
+	 * @param string $rcdata_then_div HTML with RCDATA before a DIV.
+	 * @param string $rcdata_tag      RCDATA tag.
+	 */
+	public function test_next_tag_ignores_contents_of_rcdata_tag( $rcdata_then_div, $rcdata_tag ) {
+		$p = new WP_HTML_Tag_Processor( $rcdata_then_div );
+		$p->next_tag();
+		$this->assertSame( $rcdata_tag, $p->get_tag(), "The first found tag was not '$rcdata_tag'" );
+		$p->next_tag();
+		$this->assertSame( 'DIV', $p->get_tag(), "The second found tag was not 'div'" );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data_next_tag_ignores_contents_of_rcdata_tag() {
+		return array(
+			'simple textarea'                          => array(
+				'rcdata_then_div' => '<textarea><span class="d-none d-md-inline">Back to notifications</span></textarea><div></div>',
+				'rcdata_tag'      => 'TEXTAREA',
+			),
+			'simple title'                             => array(
+				'rcdata_then_div' => '<title><span class="d-none d-md-inline">Back to notifications</title</span></title><div></div>',
+				'rcdata_tag'      => 'TITLE',
+			),
+			'comment opener inside a textarea tag should be ignored' => array(
+				'rcdata_then_div' => '<textarea class="d-md-none"><!--</textarea><div></div>-->',
+				'rcdata_tag'      => 'TEXTAREA',
+			),
+			'textarea closer with another textarea tag in closer attributes' => array(
+				'rcdata_then_div' => '<textarea><span class="d-none d-md-inline">Back to notifications</title</span></textarea <textarea><div></div>',
+				'rcdata_tag'      => 'TEXTAREA',
+			),
+			'textarea closer with attributes'          => array(
+				'rcdata_then_div' => '<textarea class="d-md-none"><span class="d-none d-md-inline">Back to notifications</span></textarea id="test"><div></div>',
+				'rcdata_tag'      => 'TEXTAREA',
+			),
+			'textarea opener with title closer inside' => array(
+				'rcdata_then_div' => '<textarea class="d-md-none"></title></textarea><div></div>',
+				'rcdata_tag'      => 'TEXTAREA',
+			),
+		);
+	}
+
+	/**
+	 * @ticket 56299
+	 *
+	 * @covers WP_HTML_Tag_Processor::next_tag
+	 *
+	 * @dataProvider data_skips_contents_of_script_and_rcdata_regions
+	 *
+	 * @param $input_html HTML with multiple divs, one of which carries the "target" attribute.
+	 */
+	public function test_skips_contents_of_script_and_rcdata_regions( $input_html ) {
+		$p = new WP_HTML_Tag_Processor( $input_html );
+		$p->next_tag( 'div' );
+
+		$this->assertTrue( $p->get_attribute( 'target' ) );
+	}
+
+	/**
+	 * Data provider
+	 *
+	 * @return array[]
```

Comment:
```suggestion
	 * @return array[] Test cases for the test_skips_contents_of_script_and_rcdata_regions method.
```

Reply:
@adamziel these were removed previously with a "for now" approach that is being used in Core. See https://github.com/WordPress/wordpress-develop/pull/3920#discussion_r1094585197. I hope you don't mind, but I'll close this as resolved.